cstar/tests/roms/fixtures.py

```diff
@@ -0,0 +1,155 @@
+       tools data directory, ensuring the data is available during testing.
+    """
+
+    def _fetch_roms_tools_source_data(symlink_path: str | Path) -> None:
+        """
+        Fetches ROMS tools source data and creates a symlink to the data directory at `symlink_path`.
+
+        This function downloads the required data files from a specified remote repository
+        to ROMS_TOOLS_DATA_DIRECTORY, specified in tests/config.py. After fetching the files,
+        it creates a symlink to the data directory at the specified `symlink_path`.
+
+        The symlink is used as the full path to the cached directory will vary from system to system, but
+        calls to roms-tools.ROMSToolsObject.from_yaml accept a yaml file with a predetermined 'path' entry.
+        This entry can be relative, so the calling test can create a symlink at a location corresponding to
+        this path, but pointing to the correct cache directory.
+
+        A cleanup function is registered to remove the symlink after the test completes.
+
+        Parameters
+        ----------
+        symlink_path : str or Path
+            The path where the symlink to the fetched data directory will be created.
+
+        Returns
+        -------
+        None
+        """
+
+        if isinstance(symlink_path, str):
+            symlink_path = Path(symlink_path)
+
+        # cache_dir = Path(pooch.os_cache("roms_tools_datasets_for_cstar_test_case"))
+        pup_test_data = pooch.create(
+            path=ROMS_TOOLS_DATA_DIRECTORY,
+            base_url="https://github.com/CWorthy-ocean/roms-tools-test-data/raw/main/",
+            # The registry specifies the files that can be fetched
+            registry={
+                "GLORYS_NA_2012.nc": "b862add892f5d6e0d670c8f7fa698f4af5290ac87077ca812a6795e120d0ca8c",
+                "ERA5_NA_2012.nc": "d07fa7450869dfd3aec54411777a5f7de3cb3ec21492eec36f4980e220c51757",
+                "TPXO_global_test_data.nc": "457bfe87a7b247ec6e04e3c7d3e741ccf223020c41593f8ae33a14f2b5255e60",
+                "CESM_BGC_2012.nc": "e374d5df3c1be742d564fd26fd861c2d40af73be50a432c51d258171d5638eb6",
+                "CESM_BGC_SURFACE_2012.nc": "3c4d156adca97909d0fac36bf50b99583ab37d8020d7a3e8511e92abf2331b38",
+            },
+        )
+        pup_test_data.fetch("GLORYS_NA_2012.nc")
+        pup_test_data.fetch("TPXO_global_test_data.nc")
+        pup_test_data.fetch("ERA5_NA_2012.nc")
+        pup_test_data.fetch("CESM_BGC_2012.nc")
+        pup_test_data.fetch("CESM_BGC_SURFACE_2012.nc")
+
+        if symlink_path.is_symlink():
+            raise FileExistsError(
+                f"{symlink_path} already exists as a symlink to {symlink_path.resolve()}"
+            )
+
+        symlink_path.symlink_to(ROMS_TOOLS_DATA_DIRECTORY)
+
+        def cleanup():
+            if symlink_path.is_symlink():
+                print(f"removing {symlink_path}")
+                symlink_path.unlink()
+
+        request.addfinalizer(cleanup)
```

Comment:
This seems like a complicated way to just end up replicating what `pytest`'s built-in `tmpdir` fixture does?

Reply:
I'm not sure if there's a way to create a temporary symlink with tmpdir but I'll happily switch to that if so

Reply:
That's certainly a clever idea, but maybe if you did what I'm suggesting in https://github.com/CWorthy-ocean/C-Star/pull/108#discussion_r1779093213 (and don't modify what's in that `tmpdir` you won't need this symlink any more?

Reply:
I believe however we handle the fetching of roms-tools data, a symlink will be needed here. The roms-tools yaml files contain a "path" entry that here is just a relative path to `roms_tools_datasets/FILE.nc` in each case. So either we have to do (more) messy yaml editing with marked template files and placeholders, or we can just make sure something is at the location expected by roms-tools (as is done with this symlink implementation). I think in this case the former is avoidable.

Reply:
Yeah editing the yamls further is messy...

> a relative path to roms_tools_datasets/FILE.nc in each case

This is relative to the `CASEROOT`? (which is a `tmpdir`?) 

We should really write out a description of what the eventual decision is here as part of some developer docs because this is confusing 😅